homeassistant/components/lastfm/config_flow.py

```diff
@@ -0,0 +1,227 @@
+                        )
+                    ),
+                }
+            ),
+        )
+
+    async def async_step_import(self, import_config: ConfigType) -> FlowResult:
+        """Import config from yaml."""
+        for entry in self._async_current_entries():
+            if entry.data[CONF_API_KEY] == import_config[CONF_API_KEY]:
+                return self.async_abort(reason="already_configured")
+        return self.async_create_entry(
+            title="LastFM",
+            data={
+                CONF_API_KEY: import_config[CONF_API_KEY],
+                CONF_MAIN_USER: None,
+                CONF_USERS: import_config[CONF_USERS],
+            },
+        )
+
+
+class LastFmOptionsFlowHandler(OptionsFlowWithConfigEntry):
+    """LastFm Options flow handler."""
+
+    def __init__(self, entry: ConfigEntry) -> None:
+        """Initialize LastFM Options flow."""
+        super().__init__(entry)
+        self.data: dict[str, Any] = dict(entry.data)
+        self.entry = entry
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Initialize form."""
+        valid_users = self.data[CONF_USERS]
+        errors = {}
+        if user_input is not None:
+            valid_users = []
+            for username in user_input[CONF_USERS]:
+                lastfm_user = get_lastfm_user(self.data[CONF_API_KEY], username)
+                lastfm_errors = validate_lastfm_user(lastfm_user)
+                if lastfm_errors:
+                    errors = lastfm_errors
+                else:
+                    valid_users.append(username)
+            if not errors:
+                return self.async_create_entry(
+                    title="LastFM",
+                    data={
+                        **self.data,
+                        CONF_USERS: user_input[CONF_USERS],
+                    },
+                )
+        if self.data[CONF_MAIN_USER]:
+            try:
+                main_user = get_lastfm_user(
+                    self.data[CONF_API_KEY], self.data[CONF_MAIN_USER]
+                )
+                friends: Sequence[SelectOptionDict] = [
+                    {"value": str(friend.name), "label": str(friend.get_name(True))}
+                    for friend in main_user.get_friends()
+                ]
+            except WSError:
+                friends = []
+        else:
+            friends = []
+        return self.async_show_form(
+            step_id="init",
+            errors=errors,
+            data_schema=self.add_suggested_values_to_schema(
+                vol.Schema(
+                    {
+                        vol.Required(CONF_USERS): SelectSelector(
+                            SelectSelectorConfig(
+                                options=friends, custom_value=True, multiple=True
+                            )
+                        ),
+                    }
+                ),
+                {CONF_USERS: valid_users},
```

Comment:
```suggestion
                user_input or self.options,
```

Reply:
So for filtering out the wrong users this would not be needed right

Reply:
Should probably be `valid_users` and not `user_input`
`valid_users or self.options`

Reply:
I think you should remove `valid_users` and instead work with removing users from `user_input[CONF_USERS]` that fail validation.
Then my initial proposal is fine as if `user_input` is None it will go for `self.options` instead
You can probably do same in `ConfigFlow` too